engine/src/main/java/com/datatorrent/stram/engine/StreamingContainer.java

```diff
@@ -769,7 +769,11 @@ public void processHeartbeatResponse(ContainerHeartbeatResponse rsp)
     }
 
     if (rsp.committedWindowId != lastCommittedWindowId) {
+
       lastCommittedWindowId = rsp.committedWindowId;
+
+      bufferServer.purge(lastCommittedWindowId);
```

Comment:
I would think it would be an option in that case too. Do you know if there was a discussion to dev list to fundamentally change the approach and only do inline. If it was discussed and this conclusion was reached I think it would be prudent to wait for that change to get in before merging this one.